### ISSUE TITLE
Logstash exits silently when there are permission issues to log file

### ISSUE THREAD

**Issue by ppf2** [elastic/logstash#5549]
Reproducible on 2.3.3.

If the -l log file has permission issues, logstash just silently exists.  Will be nice if we log something to stdout, etc..

```
touch /tmp/z
chmod 0 /tmp/z
bin/logstash -l /tmp/z --debug
# exits silently.
echo $?
1
```

**Comment by purbon** [comment]
This does not happen in master but I confirm the issue with 2.3 branch.

**Comment by purbon** [comment]
output with 5.0 is 

```
skywalker% bin/logstash -l /tmp/z        
--- jar coordinate com.fasterxml.jackson.core:jackson-annotations already loaded with version 2.7.1 - omit version 2.7.0
--- jar coordinate com.fasterxml.jackson.core:jackson-databind already loaded with version 2.7.1 - omit version 2.7.1-1
ERROR: Failed to open /tmp/z for writing: Permission denied - /tmp/z

This is often a permissions issue, or the wrong
path was specified?
usage:
  bin/logstash -f CONFIG_PATH [-t] [-r] [--quiet|verbose|debug] [-w COUNT] [-l LOG]
  bin/logstash -e CONFIG_STR [-t] [--quiet|verbose|debug] [-w COUNT] [-l LOG]
  bin/logstash -i SHELL [--quiet|verbose|debug]
  bin/logstash -V [--verbose|debug]
  bin/logstash --help
skywalker% 
```

so the error is properly handled there.

**Comment by suyograo** [comment]
Fixed in #5587
